In [1]:
import os
import shutil
import pandas as pd

In [12]:
csv_path = "prob.csv"  
image_folder = r"C:\Users\PC\OneDrive\Escritorio\PUCP\2025-0\DEEP LEARNING\PROYECTO\images_training_rev1\images_training_rev1"  
output_folder = "training_set"  

df = pd.read_csv(csv_path)  # Leer CSV
df = df.dropna(subset=["class"])  # Asegurar que no haya NaN en la columna "class"
df

GalaxyID       class
0        100008      sphere
1        100078      sphere
2        100090      sphere
3        100122  elliptical
4        100123      sphere
...         ...         ...
43210    999936  elliptical
43211    999950  elliptical
43212    999958  elliptical
43213    999964     espiral
43214    999967  elliptical

[43215 rows x 2 columns]

In [10]:
categories = ["elliptical", "espiral", "sphere", "irregular"]
for category in categories:
    os.makedirs(os.path.join(output_folder, category), exist_ok=True)

for _, row in df.iterrows():
    image_id = str(row["GalaxyID"])
    category = row["class"]  
    
    source_path = os.path.join(image_folder, image_id + ".jpg")
    destination_path = os.path.join(output_folder, category, image_id + ".jpg")
    
    # Mover archivo si existe
    if os.path.exists(source_path):
        shutil.move(source_path, destination_path)
    else:
        print(f" Imagen no encontrada: {source_path}") 
print("✅ Organización de imágenes completada.")


✅ Organización de imágenes completada.


In [11]:
15109 + 8314 + 2497 + 17295

43215

registros totales:  61578
registros 'espirales':  8314
registros 'sphere':  17295
registros 'elliptical':  15109
registros 'irregular':  2497
registros asignados:  43215
registros faltantes:  18363

In [15]:
print("SE NECESITAN ",17295 - 15109," IMAGENES DE GALAXIAS ELIPTICAS")
print("SE NECESITAN ",17295 - 8314," IMAGENES DE GALAXIAS ESPIRALES")
print("SE NECESITAN ",17295 - 2497," IMAGENES DE GALAXIAS IRREGULARES")

SE NECESITAN  2186  IMAGENES DE GALAXIAS ELIPTICAS
SE NECESITAN  8981  IMAGENES DE GALAXIAS ESPIRALES
SE NECESITAN  14798  IMAGENES DE GALAXIAS IRREGULARES


In [16]:
17295*4

69180

## Posterior al Data Augmentation, TRAIN VALID Y TEST

In [1]:
import os
import shutil
import random

In [4]:
base_dir = r"C:\Users\PC\OneDrive\Escritorio\PUCP\2025-0\DEEP LEARNING\PROYECTO\MODELO\Galaxy Classifier master\training_set"
output_dirs = ["train", "valid", "test"]
for folder in output_dirs:
    os.makedirs(folder, exist_ok=True)
categories = [d for d in os.listdir(base_dir) if os.path.isdir(os.path.join(base_dir, d))]
categories

['elliptical', 'espiral', 'irregular', 'sphere']

In [5]:
# Recorre cada categoría (clase de galaxia)
for category in categories:
    category_path = os.path.join(base_dir, category)
    images = os.listdir(category_path)

    # Mezclar imágenes aleatoriamente
    random.shuffle(images)

    # Definir tamaños de cada conjunto
    total_images = len(images)
    train_count = int(0.70 * total_images)
    valid_count = int(0.15 * total_images)

    train_images = images[:train_count]
    valid_images = images[train_count:train_count + valid_count]
    test_images = images[train_count + valid_count:]

    # Crear carpetas de categorías en train, valid, test
    for folder in output_dirs:
        os.makedirs(os.path.join(folder, category), exist_ok=True)

    # Función para mover imágenes
    def move_images(image_list, dest_folder):
        for img in image_list:
            src_path = os.path.join(category_path, img)
            dst_path = os.path.join(dest_folder, category, img)
            shutil.move(src_path, dst_path)

    # Mover imágenes
    move_images(train_images, "train")
    move_images(valid_images, "valid")
    move_images(test_images, "test")

    print(f"{category}: {len(train_images)} en train, {len(valid_images)} en valid, {len(test_images)} en test.")

print("✅ Distribución completada.")


elliptical: 12106 en train, 2594 en valid, 2595 en test.
espiral: 12106 en train, 2594 en valid, 2595 en test.
irregular: 12106 en train, 2594 en valid, 2595 en test.
sphere: 12106 en train, 2594 en valid, 2595 en test.
✅ Distribución completada.


In [1]:
print("SE NECESITAN ",17295 - 15109," IMAGENES DE GALAXIAS ELIPTICAS")
print("SE NECESITAN ",17295 - 8314," IMAGENES DE GALAXIAS ESPIRALES")
print("SE NECESITAN ",17295 - 2497," IMAGENES DE GALAXIAS IRREGULARES")

SE NECESITAN  2186  IMAGENES DE GALAXIAS ELIPTICAS
SE NECESITAN  8981  IMAGENES DE GALAXIAS ESPIRALES
SE NECESITAN  14798  IMAGENES DE GALAXIAS IRREGULARES


In [2]:
12106-1720

10386

## REGRESAR LA DATA A LA CARPETA ORIGINAL (100%)

## DIVIDIR LA DATA EN 3 CARPETAS (BRUNO, FIORELLA Y JEREMY)

In [5]:
import os
import shutil
import random

# Definir los nombres de los nuevos conjuntos
personas = ["bruno", "fiorella", "jeremy"]
datasets = ["training_set", "val_set", "test"]
categorias = ["elliptical", "sphere", "espiral", "irregular"]


# Función para dividir imágenes en 3 partes
def dividir_imagenes(origen, destino_base):
    if not os.path.exists(origen):
        print(f"❌ ERROR: La carpeta '{origen}' no existe.")
    for categoria in categorias:
        ruta_categoria = os.path.join(origen, categoria)
        imagenes = os.listdir(ruta_categoria)
        random.shuffle(imagenes)  # Mezclar aleatoriamente
        
        # Dividir en 3 partes
        tam_parte = len(imagenes) // 3
        partes = [imagenes[:tam_parte], imagenes[tam_parte:2*tam_parte], imagenes[2*tam_parte:]]
        
        # Crear carpetas para cada persona y mover imágenes
        for i, persona in enumerate(personas):
            destino = os.path.join(destino_base, persona, os.path.basename(origen), categoria)
            os.makedirs(destino, exist_ok=True)
            
            for img in partes[i]:
                shutil.copy(os.path.join(ruta_categoria, img), os.path.join(destino, img))

# Aplicar la función a cada dataset
for dataset in datasets:
    dividir_imagenes(dataset, "./")  # "./" indica que se crea en el mismo directorio de trabajo

print("✅ División completada.")


✅ División completada.


## DATA CORREGIDA (20/02/2025)

In [5]:
import os
import shutil

carpetas_originales = ["training_set", "test", "val_set"]
carpetas_corregidas = ["training_set_corregido", "test_corregido", "val_set_corregido"]

for original, corregido in zip(carpetas_originales, carpetas_corregidas):
    if os.path.exists(original):  
        shutil.copytree(original, corregido)
        print(f"Copia creada: {corregido}")
    else:
        print(f"La carpeta {original} no existe en el directorio actual.")

print("✅ ¡Copias completadas!")


Copia creada: training_set_corregido
Copia creada: test_corregido
Copia creada: val_set_corregido
✅ ¡Copias completadas!
